blog/2020-11-19-project-droplet-closing-notes.md

```diff
@@ -0,0 +1,187 @@
+A more detailed explanation would go beyond the scope of this blog post. 
+We refer the interested reader to [[8]](https://elementsproject.org/features/confidential-transactions/investigation) and our PoC in [[3]](https://github.com/comit-network/droplet).
+
+### Native assets
+
+Another core feature of Elements is the ability to issue assets. 
+Every asset is identified by an `Asset ID` and has one (or a federation of multiple) issuer. 
+A list of all currently available assets on Elements can be found here [[7]](https://blockstream.info/liquid/assets). 
+In our opinion the most interesting assets on Liquid are L-BTC - a token pegged 1-to-1 to the Bitcoin mainchain - and L-USDT - a USD stable coin issued by Tether.
+
+When transferring any asset the sender has to specify the `asset` ID she wishes to transfer.
+
+Note: Transaction fees on Liquid are always paid in L-BTC. 
+However, one can use a third party such as Liquid Taxi [[5]](https://liquid.taxi/) who will pay for your transaction fees in L-BTC. 
+In return you pay the service provider in a different asset.
+
+A non-confidential UTXO can look like this:
+
+```json
+{
+      "value": 10.00000000,
+      "asset": "b2e15d0d7a0c94e4e2ce0fe6e8691b9e451377f6e46e8045a86f7c4b5d4f0f23",
+      "commitmentnonce": "",
+      "commitmentnonce_fully_valid": false,
+      "n": 1,
+      "scriptPubKey": {
+        "asm": "OP_HASH160 654af855ed5bb080d9597e8fe215c61a52ac3d40 OP_EQUAL",
+        "hex": "a914654af855ed5bb080d9597e8fe215c61a52ac3d4087",
+        "reqSigs": 1,
+        "type": "scripthash",
+        "addresses": [
+          "XLapuXkFSue2nhrj4eLiAB6MdA32eEmcE2"
+        ]
+      }
+    }
+
+```
+
+### Swap protocol
+
+Same as Bitcoin, Elements' transactions follow the UTXO model. 
+With that in mind, we can build a true atomic swap protocol which allows for swapping two assets within a single transaction. 
+If this transaction is mined, the swap was successful, if not, the swap didn't happen and no transaction fees were wasted. 
+No hashlocks, timelocks, or refund transactions are needed. 
+
+Our swap protocol assumes that the two parties have already negotiated what amounts and asset types they want to exchange. We also assume that one of the assets is L-BTC, so that the party selling it can take care of the transaction fees. 
+In order to collaboratively create a valid transaction which exchanges the ownership of these 2 assets, the two parties need to share some additional information. 
+This is what the protocol looks like:
+
+#### Step 1 
+
+Alice sends her `input` to Bob, identifying which UTXO Alice will be spending from. 
+If this input is confidential, she also has to provide the input's blinding key `input_blinding_sk`. 
+Importantly, this is _not_ the private key for the address she is spending from! With this key, Bob is only able to unblind the input and see its value and asset type. 
```

Comment:
```suggestion
Importantly, this is _not_ the private key for the address she is spending from!
With this key, Bob is only able to unblind the input and see its value and asset type. 
```

Reply:
Is `unblind` the best available terminology?

Reply:
I think so, it's also used in the RPC Api and crypto libraries we use. :)